string/string.mbt

```diff
@@ -111,3 +111,42 @@ test "to_array" {
   inspect(a[4], content="b")?
   inspect(a[5], content="t")?
 }
+
+/// Returns an integer between 0 and 65535 representing the UTF-16 code unit at the given index.
+pub fn char_code_at(self : String, index : Int) -> Option[Int] {
+  if index < 0 || index >= self.length() {
+    None
+  } else {
+    Some(self[index].to_int())
+  }
+}
+
+test "char_code_at" {
+  inspect("ABC".char_code_at(0), content="Some(65)")?
+  inspect("🤣".char_code_at(0), content="Some(55358)")?
+  inspect("🤣".char_code_at(1), content="Some(56611)")?
```

Comment:
char_code_at and code_point_at seems to be inconsistent with String.prototype.charCodeAt and String.prototype.codePointAt, is this intentional?

Reply:
In javascript, if the index out of range, charCodeAt will return a `NaN`, and codePointAt will return an `undefined`. How can we approach this behavior in MoonBit?

Reply:
https://github.com/moonbitlang/core/issues/299
I wrote some of my thoughts here